components/raftstore-v2/src/raft/peer.rs

```diff
@@ -27,6 +52,12 @@ pub struct Peer<EK: KvEngine, ER: RaftEngine> {
     pub(crate) async_writer: AsyncWriter<EK, ER>,
     has_ready: bool,
     pub(crate) logger: Logger,
+    pub(crate) pending_reads: ReadIndexQueue<QueryResChannel>,
```

Comment:
Should use accessor just like other fields instead of making them public. There are several fields are marked as `pub(crate)` because they can lead to borrow check failure.

Reply:
> There are several fields are marked as pub(crate) because they can lead to borrow check failure.
Though they have valid reason, still they look very inconsistent.  And in fact, pending_reads also has a valid reason.
self.pending_reads_mut().advance_leader_reads(self.tag(), states)---it won't compile. 

The bigger problem I think in fact is splitting Peer's implementation to different files are not very right. These functions are very tight to the core of peer.rs and thus they need to access every private field of Peer's definition. But now they have to access them through accessors, which essentially make these field pub. It breaks the encapsulation principle already.

Reply:
`region_id` and `peer_id` should be used in `ReadIndexQueue`.

Reply:
used tag to make it compatible with v1's code. As discussed, v2's peer does not have tag field.